ee/src/main/java/org/jboss/as/ee/component/deployers/EECleanUpProcessor.java

```diff
@@ -46,6 +48,7 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
 
     @Override
     public void undeploy(final DeploymentUnit context) {
-
+        final Module module = context.getAttachment(org.jboss.as.server.deployment.Attachments.MODULE);
+        FactoryFinderCache.clearClassLoader(module.getClassLoader());
```

Comment:
@stuartwdouglas wdyt about this?

We already have similar cache clear for web deployments, but in case of bean validation EL can also get used and as such expression language factories loaded and classloaders cached.

I am not really sure this is the right place for such fix.